### PR TITLE
Adds copy template option #148

### PR DESCRIPTION
**Pull request by kmcgre**

### CODE REVIEW

**Comment by xsrust** [comment]
The logic all looks good to me.  It would be useful to confirm with @briri that the error messaging follows the new formatting he has implimented in his work last week.

File: app/controllers/templates_controller.rb [hunk 1]
```diff
@@ -323,4 +323,29 @@ def admin_template_history
...
+    new_copy.dmptemplate_id = loop do
+      random = rand 2147483647
+      break random unless Template.exists?(dmptemplate_id: random)
+    end
+
+    if new_copy.save
+      flash[:notice] = 'Template was successfully copied.'
+      redirect_to admin_template_template_path(id: new_copy.id, edit: true), notice: _('Information was successfully created.')
+    else
+      flash[:notice] = failed_create_error(new_copy, _('template'))
```
Comment (briri) [inline app/controllers/templates_controller.rb]: Change this to `flash[:alert]`. We have a div on the main `layouts/application.html.erb` that will display this and has all of the aria markup to announce it correctly to screen readers.

**Comment by briri** [comment]
Just update that one line. Otherwise everything looks good!

File: app/controllers/templates_controller.rb [hunk 2]
```diff
@@ -323,4 +323,29 @@ def admin_template_history
     @current = Template.current(@template.dmptemplate_id)
   end
 
+  # PUT /org/admin/templates/:id/admin_copy
+  # -----------------------------------------------------
+  def admin_copy
+    @template = Template.find(params[:id])
+    authorize @template
+
+    # Unpublish the live version
```
Comment (vyruss) [inline app/controllers/templates_controller.rb]: Perhaps remove this comment line as it may be confusing

**Pull request merged**